table/block_based_table_reader.cc

```diff
@@ -1230,32 +1235,35 @@ Status BlockBasedTable::GetDataBlockFromCache(
 
   // found compressed block
   RecordTick(statistics, BLOCK_CACHE_COMPRESSED_HIT);
-  compressed_block = reinterpret_cast<Block*>(
+  compressed_block = reinterpret_cast<BlockContents*>(
       block_cache_compressed->Value(block_cache_compressed_handle));
```

Comment:
So block_cache_compressed will store BlockContent instead of Block? This would be a major change? Where is that we are documenting this change?

Reply:
See the PR title and summary. I updated it after I added the change.